addons/web/models/models.py

```diff
@@ -983,10 +983,8 @@ def onchange2(self, values: Dict, field_names: List[str], fields_spec: Dict):
         # values of the move, among which the one2many that contains the line
         # itself, with old values!
         #
-        changed_values = {fname: values[fname] for fname in field_names}
-        # set changed values to null in initial_values; not setting them
-        # triggers default_get() on the new record when creating snapshot0
-        initial_values = dict(values, **dict.fromkeys(field_names, False))
+        initial_values = dict(values)
+        changed_values = {fname: initial_values.pop(fname) for fname in field_names}
```

Comment:
still a bit (too) complicated :(

Reply:
the `initial_values` for `create`d record in the next loop at

```py
for field_name in initial_values:
    field = self._fields.get(field_name)
    if field and field.inherited:
        parent_name, field_name = field.related.split('.', 1)
record[parent_name]._update_cache({field_name: record[field_name]})
```

is now only done on the values from the web client instead of *all* values (a bit better)